### DISCUSSION TITLE
One view for all records

### DISCUSSION THREAD

**Discussion by jhofmueller** [sys4/netbox-plugin-dns#252] [Ideas]
Currently there are two views: one for records and one for managed records. Since both are of the same type it would be consistent to show then in one view. Managed records cannot be edited, so this already marks a difference. One could also add a default column that displays if the record is manage or not.

There is a use case where the current solution could create confusion. In one of our tests we have tenants that have managed records. They appear in the tenant's "Related Objects" overview. Clicking on "Records" in this overview takes one to the "Records" menu item which does *not* show the tenant's records.

Just an idea :sunglasses:

**Comment by peteeckel** [comment]
Hi @jhofmueller, thanks for the input.

The problem with having two views for records and managed records is that the tables for both need to be different.

Because you can't edit or delete managed records, having them in a table with edit/delete buttons on the right of each column is at least confusing and will lead to tons of support issues ... until now I didn't find a way to display specific buttons for each row, and even that would be confusing for many users. So it's more a UX issue than anything else.

And yes, I discovered the problem with the 'Tenants' overview as well - so far I don't have a good idea how to fix that, but it's still less pain than the table issue with managed/unmanaged records.

I already made a mental note to revisit that stuff once I have the time, but will add it to my backlog as well so I don't lose track. When/if I find a good solution (or if someone is willing to contribute one ... hint, hint) I will implement it.

**Reply by jhofmueller** [reply]
Hi @peteeckel!

I see what you mean. Just experienced something similar with a user that has view rights on an object but sees and edit button in the list of objects and gets an error when clicking it ...

Thanks for your effort!

**Comment by bitcollector1** [comment]
I actually like having them in different tables since I manually define all my infrastructure boxes and use managed records for everything else. I just like the separation it allows for the critical records.
